### PR TITLE
Laravel: init at 5.8.1

### PR DESCRIPTION
**Pull request by heisfer**
## Description of changes
A CLI tool to create Laravel projects.

GitHub repo: https://github.com/laravel/installer
Docs: https://laravel.com/docs#creating-a-laravel-project
Formatter: nixfmt-rfc-style

(You still need to install composer separately in order for this to work).

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/la/laravel/package.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    owner = "laravel";
+    repo = "installer";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-LQABZnmKgJ8qkymmSjhjc+x1qZ/tFqFyQbfeGZECxok=";
+  };
+
+  composerLock = ./composer.lock;
+  vendorHash = "sha256-f18N2qNCUFetCaHaC4X6Benq70x21SVQ3YSs8kovK1g=";
+
+  meta = with lib; {
```
Comment (isabelroses) [inline pkgs/by-name/la/laravel/package.nix:20]: `with lib;` is an anti-pattern, see here https://github.com/NixOS/nixpkgs/issues/208242.

Comment (heisfer) [reply pkgs/by-name/la/laravel/package.nix:20]: maintainers = with lib.maintainers; [ <someone> ]: is fine?

Comment (isabelroses) [reply pkgs/by-name/la/laravel/package.nix]: Yeah that's fine.

**Review by isabelroses — APPROVED** [review]

**Review by yayayayaka — COMMENTED** [review]
> (You still need to install composer separately in order for this to work).

I think that could be solved by wrapping the program. Tried it out by doing https://github.com/yayayayaka/nixpkgs/commit/0e5be5f1ac41d3225cfab827c8840148c45bac86 and seems to work.

**Comment by heisfer** [comment]
> I think that could be solved by wrapping the program. Tried it out by doing [yayayayaka@0e5be5f](https://github.com/yayayayaka/nixpkgs/commit/0e5be5f1ac41d3225cfab827c8840148c45bac86) and seems to work.

Not a bad idea. My first thought was that they install whatever composer version they want, but in the end it's just Laravel application installer, so I'll add the change. I also tested the code and seems to work just fine.

**Review by yayayayaka — APPROVED** [review]
Diff looking fine and tested on x86_64-linux by creating a laravel project and starting the development server.

**Review by nicoonoclaste — APPROVED** [review]

**Pull request merged**